dotnet/src/Connectors/Connectors.OpenAIV2/Core/ClientCore.Embeddings.cs

```diff
@@ -0,0 +1,64 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+/* 
+Phase 01
+
+This class was created to simplify any Text Embeddings Support from the v1 ClientCore
+*/
+
+using System;
+using System.ClientModel;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using OpenAI.Embeddings;
+
+#pragma warning disable CA2208 // Instantiate argument exceptions correctly
+
+namespace Microsoft.SemanticKernel.Connectors.OpenAI;
+
+/// <summary>
+/// Base class for AI clients that provides common functionality for interacting with OpenAI services.
+/// </summary>
+internal partial class ClientCore
+{
+    /// <summary>
+    /// Generates an embedding from the given <paramref name="data"/>.
+    /// </summary>
+    /// <param name="data">List of strings to generate embeddings for</param>
+    /// <param name="kernel">The <see cref="Kernel"/> containing services, plugins, and other state for use throughout the operation.</param>
+    /// <param name="dimensions">The number of dimensions the resulting output embeddings should have. Only supported in "text-embedding-3" and later models.</param>
+    /// <param name="cancellationToken">The <see cref="CancellationToken"/> to monitor for cancellation requests. The default is <see cref="CancellationToken.None"/>.</param>
+    /// <returns>List of embeddings</returns>
+    internal async Task<IList<ReadOnlyMemory<float>>> GetEmbeddingsAsync(
+        IList<string> data,
+        Kernel? kernel,
+        int? dimensions,
+        CancellationToken cancellationToken)
```

Comment:
I think this method should have an execution settings TBH where the ModelId would be overriden per request, I will keep track on this ask to investigate further.

Investigation: Check if this would require rethinking on `ITextEmbeddingsGeneration` interface which could have bigger consequences, impacting other connectors implementations, as well as how this also relates to the work on the Vector abstractions @westey-m is doing.